interp/interp_test.go

```diff
@@ -191,3 +194,228 @@ func TestPiecewiseConstantPredict(t *testing.T) {
+			dydxs:           []float64{0, 0, 0},
+			expectedMessage: differentLengths,
+		},
+		{
+			xs:              []float64{0, 1, 1},
+			ys:              []float64{10, 20, 30},
+			dydxs:           []float64{0, 0, 0, 0},
+			expectedMessage: differentLengths,
+		},
+		{
+			xs:              []float64{0},
+			ys:              []float64{0},
+			dydxs:           []float64{0},
+			expectedMessage: tooFewPoints,
+		},
+		{
+			xs:              []float64{0, 1, 1},
+			ys:              []float64{10, 20, 10},
+			dydxs:           []float64{0, 0, 0},
+			expectedMessage: xsNotStrictlyIncreasing,
+		},
+	} {
+		var pc PiecewiseCubic
+		err := pc.FitWithDerivatives(test.xs, test.ys, test.dydxs)
+		if err == nil || err.Error() != test.expectedMessage {
+			t.Errorf("expected error for xs: %v, ys: %v and dydxs: %v with message: %s", test.xs, test.ys, test.dydxs, test.expectedMessage)
+		}
+	}
+}
+
+func leftPoly(x float64) float64 {
+	return x*x - x + 1
+}
+
+func leftPolyDerivative(x float64) float64 {
+	return 2*x - 1
+}
+
+func rightPoly(x float64) float64 {
+	return x*x*x - x + 1
+}
+
+func rightPolyDerivative(x float64) float64 {
+	return 3*x*x - 1
+}
+
+func testFittedPolys(t *testing.T, pc *PiecewiseCubic) {
+	lastY := rightPoly(1)
+	for i, test := range []struct {
+		x    float64
+		want float64
+	}{
+		{-2, 3},
+		{-1, 3},
+		{-0.9, leftPoly(-0.9)},
+		{-0.75, leftPoly(-0.75)},
+		{-0.5, leftPoly(-0.5)},
+		{-0.25, leftPoly(-0.25)},
+		{-0.1, leftPoly(-0.1)},
+		{0, 1},
+		{0.1, rightPoly(0.1)},
+		{0.25, rightPoly(0.25)},
+		{0.5, rightPoly(0.5)},
+		{0.75, rightPoly(0.75)},
+		{0.9, rightPoly(0.9)},
+		{1, lastY},
+		{2, lastY},
+	} {
+		got := pc.Predict(test.x)
+		if math.Abs(got-test.want) > 1e-14 {
+			t.Errorf("Mismatch in test case %d for x = %g: got %v, want %g", i, test.x, got, test.want)
+		}
+	}
+}
+
+func TestAkimaSplinesSingleFunction(t *testing.T) {
+	t.Parallel()
+	const (
+		nPts = 40
+		tol  = 1e-14
+	)
+	for i, test := range []struct {
+		xs, ys []float64
+		f      func(float64) float64
+	}{
+		{
+			xs: []float64{-1, 0, 1},
+			ys: []float64{1, 0, 1},
+			f:  func(x float64) float64 { return x * x },
+		},
+		{
+			xs: []float64{-1, 1},
+			ys: []float64{10, -10},
+			f:  func(x float64) float64 { return -10 * x },
+		},
+		{
+			xs: []float64{-1, 0, 1},
+			ys: []float64{10, 0, -10},
+			f:  func(x float64) float64 { return -10 * x },
+		},
+	} {
+		var as AkimaSplines
+		err := as.Fit(test.xs, test.ys)
+		if err != nil {
+			t.Errorf("Error when fitting AkimaSplines in test case %d: %v", i, err)
+		}
+		x0 := test.xs[0]
+		x1 := test.xs[len(test.xs)-1]
+		dx := (x1 - x0) / nPts
+		for j := -1; j <= nPts+1; j++ {
+			x := x0 + float64(j)*dx
+			got := as.Predict(x)
+			want := test.f(math.Min(x1, math.Max(x0, x)))
+			if math.Abs(got-want) > tol {
+				t.Errorf("Mismatch in interpolated value at x == %g for test case %d: got %v, want %g", x, i, got, want)
+			}
+		}
+	}
+}
+
+func TestAkimaSplinesNoWiggles(t *testing.T) {
```

Comment:
I think this can be part of `TestAkimaSpline`.

Reply:
OK.